src/service.ts

```diff
@@ -293,32 +298,31 @@ export class Service {
     return JSON.parse(await response.text());
   }
 
-  static saveJSON(json: object, uri: string): Promise<string> {
+  static saveJSON(json: object, uri: string): any {
     const update = !!uri;
-    return new Promise((resolve, reject) => {
-      const xhr = new XMLHttpRequest();
-      xhr.addEventListener("load", function() {
-        if (update) {
-          resolve(uri);
-        } else {
-          let jsonURI = JSON.parse(this.response).uri;
-          jsonURI = jsonURI.substring(jsonURI.lastIndexOf("/") + 1);
-          resolve(jsonURI);
-        }
-      });
-      xhr.addEventListener("error", function() {
-        reject();
-      });
-      if (update) {
-        xhr.open("PUT", "//api.myjson.com/bins/" + uri, true);
-      } else {
-        xhr.open("POST", "//api.myjson.com/bins", true);
-      }
-      xhr.setRequestHeader("Content-type", "application/json; charset=utf-8");
-      xhr.send(JSON.stringify(json));
-    });
+    const url = "//api.myjson.com/bins/" + uri;
+    const url1 = "//api.myjson.com/bins/";
+    if(update){
+      fetch(url,{
+        method : "put",
+        headers: new Headers({ "Content-type": "application/json; charset=utf-8" })
+      })
+      .then(uri => {return Promise.resolve(uri)})
+    }
+    else {
+      fetch(url1, {
+        method: "post",
+        headers: new Headers({ "Content-type": "application/json; charset=utf-8" }),
+        body : JSON.stringify(json)
+      })
+      .then(response => {
+      let jsonURI = response.text();
+      // jsonURI = jsonURI.substring(jsonURI.lastIndexOf("/") + 1);
```

Comment:
How to Deal with it?

Reply:
> How to Deal with it?

I don't understand the question. Commented code is either removed or made into working code.